### PR TITLE
Remove threshold based spilling from HashBuild join

### PR DESCRIPTION
**Pull request by tanjialiang**
- Removes threshold based spilling from HashBuild operator
- Changed HashJoinTest to accommodating the deprecation of threshold based spilling

### CODE REVIEW

File: velox/exec/SharedArbitrator.cpp [hunk 1]
```diff
@@ -336,6 +336,25 @@ bool SharedArbitrator::handleOOM(
   return true;
 }
 
+uint64_t SharedArbitrator::shrinkCapacity(
```
Comment (xiaoxmeng) [inline velox/exec/SharedArbitrator.cpp:339]: Can you make this a separate change with test? Thanks!

File: velox/exec/tests/utils/OperatorTestBase.cpp [hunk 2]
```diff
@@ -32,6 +30,10 @@
 #include "velox/parse/TypeResolver.h"
 #include "velox/serializers/PrestoSerializer.h"
 
+DEFINE_bool(
```
Comment (xiaoxmeng) [inline velox/exec/tests/utils/OperatorTestBase.cpp:33]: Let's make this a separate change with default to be true

File: velox/exec/Spill.h [hunk 3]
```diff
@@ -434,6 +434,41 @@ class SpillState {
 /// Generate partition id set from given spill partition set.
 SpillPartitionIdSet toSpillPartitionIdSet(
     const SpillPartitionSet& partitionSet);
+
+/// Testing utility for spills. This utility is only used in tests.
```
Comment (xiaoxmeng) [inline velox/exec/Spill.h:438]: Let's make this change a separate change?

File: velox/exec/HashBuild.cpp [hunk 4]
```diff
@@ -548,17 +514,15 @@ void HashBuild::spillInput(const RowVectorPtr& input) {
   computeSpillPartitions(input);
 
   vector_size_t numSpillInputs = 0;
-  for (auto row = 0; row < numInput; ++row) {
-    const auto partition = spillPartitions_[row];
-    if (FOLLY_UNLIKELY(!activeRows_.isValid(row))) {
+  for (auto rowIdx = 0; rowIdx < numInput; ++rowIdx) {
```
Comment (xiaoxmeng) [inline velox/exec/HashBuild.cpp:517]: The other place also use row for this. Let's keep the original variable naming? Thanks!

File: velox/exec/HashBuild.cpp [hunk 5]
```diff
@@ -548,17 +514,15 @@ void HashBuild::spillInput(const RowVectorPtr& input) {
...
+    const auto partition = spillPartitions_[rowIdx];
+    if (FOLLY_UNLIKELY(!activeRows_.isValid(rowIdx))) {
       continue;
     }
-    if (!spiller_->isSpilled(partition)) {
-      continue;
-    }
-    activeRows_.setValid(row, false);
+    activeRows_.setValid(rowIdx, false);
```
Comment (xiaoxmeng) [inline velox/exec/HashBuild.cpp:522]: Let's do not mix code refactoring in this PR? Thanks!

File: velox/exec/HashBuild.cpp [hunk 6]
```diff
@@ -538,6 +502,8 @@ bool HashBuild::reserveMemory(const RowVectorPtr& input) {
 void HashBuild::spillInput(const RowVectorPtr& input) {
   VELOX_CHECK_EQ(input->size(), activeRows_.size());
 
+  // Q(jtan6): Why do we have !spiller_->isAnySpilled() condition here? If
```
Comment (xiaoxmeng) [inline velox/exec/HashBuild.cpp:505]: Yes, if nothing has been spilled, then we don't spill the input. Can you remove the question note? Thanks!

File: velox/exec/tests/HashJoinTest.cpp [hunk 7]
```diff
@@ -200,6 +203,12 @@ using JoinResultsVerifier =
 
 class HashJoinBuilder {
  public:
+  enum SpillOption {
```
Comment (xiaoxmeng) [inline velox/exec/tests/HashJoinTest.cpp:206]: Why we need this? If it is for test refactoring, let's keep it separate? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@tanjialiang thanks % comments.

File: velox/exec/HashBuild.h [hunk 8]
```diff
@@ -248,14 +224,8 @@ class HashBuild final : public Operator {
 
   std::shared_ptr<HashJoinBridge> joinBridge_;
 
-  // The maximum memory usage that a hash build can hold before spilling.
-  // If it is zero, then there is no such limit.
-  const uint64_t spillMemoryThreshold_;
-
   bool exceededMaxSpillLevelLimit_{false};
 
-  std::shared_ptr<SpillOperatorGroup> spillGroup_;
```
[the remark this answers is not in the document]
Comment (tanjialiang) [reply velox/exec/HashBuild.h:257]: Yeah this will be done after spill group is removed from aggregation

File: velox/exec/Operator.h [hunk 9]
```diff
@@ -707,7 +707,7 @@ class Operator : public BaseRuntimeStatWriter {
   /// operator if it happens to be suspended for memory arbitration processing.
   /// This only applies to a reclaimable operator.
   tsan_atomic<bool> nonReclaimableSection_{false};
-
+ public:
```
Comment (xiaoxmeng) [inline velox/exec/Operator.h:710]: Remove?

File: velox/exec/tests/HashJoinTest.cpp [hunk 10]
```diff
@@ -33,6 +34,8 @@
 #include "velox/exec/tests/utils/VectorTestUtil.h"
 #include "velox/vector/fuzzer/VectorFuzzer.h"
 
+DECLARE_bool(velox_testing_enable_arbitration);
```
Comment (xiaoxmeng) [inline velox/exec/tests/HashJoinTest.cpp:37]: No need velox_testing_enable_arbitration is declared in operator test base header file.

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang LGTM. Thanks!

**Pull request closed without merging**